packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -442,15 +454,19 @@ fn process_fn_items(
             fn selector_to_type_id(&self, sel: u64) -> usize {
                 match sel {
                     #(#abi_selectors)*
-                    // TODO: should handle this a little more gently
-                    _ => panic!("Unknown type id."),
+                    _ => {
+                        Logger::warn("Unknown selector; check ABI to make sure function outputs match to types");
```

Comment:
- Same here.
- In all these cases we want the best/most-informative error possible -- but still wanna panic.